components/camel-aws/camel-aws2-kinesis/src/main/java/org/apache/camel/component/aws2/kinesis/Kinesis2Producer.java

```diff
@@ -47,11 +53,28 @@ public Kinesis2Endpoint getEndpoint() {
 
     @Override
     public void process(Exchange exchange) throws Exception {
+        Exception exceptionThrownByPutRecord = this.exceptionThrownByPutRecord;
```

Comment:
This is not as good as concurrent calls could lead to thread safety issues - Use AtomicInstance or what the name is.

Also its not understandable why a previous exception should cause the next to fail. There is no batch stuff here. 

I think you should seperate the items you try to help with

- batching (make this work with current client)
- async (much harder - wait with this as its not needed asap)

Reply:
Hi Claus, thanks for the comments. This PR keeps using current client and supports sending batch. 
Since sending batch already has a good throughput, and as you said async process is complex, It seems that we don't really need an async client.